### PR TITLE
webgpu: implement compilation info

### PR DESCRIPTION
**Pull request by edevil**
Added getCompilationInfo() to shader object so that compilation errors can be analysed.

### CODE REVIEW

File: src/workerd/api/gpu/gpu-shader-module.c++ [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include "gpu-shader-module.h"
+
+namespace workerd::api::gpu {
+
+jsg::Promise<jsg::Ref<GPUCompilationInfo>>
+GPUShaderModule::getCompilationInfo() {
```
Comment (jasnell) [inline src/workerd/api/gpu/gpu-shader-module.c++:10]:
```suggestion
jsg::Promise<jsg::Ref<GPUCompilationInfo>> GPUShaderModule::getCompilationInfo() {
```

File: src/workerd/api/gpu/gpu-shader-module.h [hunk 2]
```diff
@@ -4,20 +4,82 @@
...
+    case WGPUCompilationMessageType_Info:
+      return kj::str("info");
+    default:
+      KJ_UNREACHABLE
+    }
+  }
+  uint64_t getLineNum() { return message.lineNum; }
+  uint64_t getLinePos() { return message.linePos; }
+  uint64_t getOffset() { return message.offset; }
+  uint64_t getLength() { return message.length; }
```
Comment (jasnell) [inline src/workerd/api/gpu/gpu-shader-module.h:47]: Just confirming... these properties are expected to be `bigint` in JavaScript? That's what `uint64_t` will be mapped to so just confirming.

File: src/workerd/api/gpu/gpu-shader-module.h [hunk 3]
```diff
@@ -4,20 +4,82 @@
...
+    JSG_READONLY_PROTOTYPE_PROPERTY(messages, getMessages);
+  }
+
+private:
+  kj::Vector<jsg::Ref<GPUCompilationMessage>> messages_;
+  kj::ArrayPtr<jsg::Ref<GPUCompilationMessage>> getMessages() {
+    return messages_.asPtr();
+  };
+  void visitForGc(jsg::GcVisitor& visitor) {
+    for (auto& message : messages_) {
```
Comment (jasnell) [inline src/workerd/api/gpu/gpu-shader-module.h:65]: There should be a `visitor.visitAll(...)`

File: src/workerd/api/gpu/gpu-shader-module.c++ (hunk 1 above)
Comment (edevil) [reply src/workerd/api/gpu/gpu-shader-module.c++:10]: Configured clang format for a 100 column limit.

File: src/workerd/api/gpu/gpu-shader-module.h (hunk 2 above)
Comment (edevil) [reply src/workerd/api/gpu/gpu-shader-module.h:47]: In the [spec](https://www.w3.org/TR/webgpu/#dom-gpucompilationmessage-linenum) they have the type [unsigned long long](https://webidl.spec.whatwg.org/#idl-unsigned-long-long).

Comment (edevil) [reply src/workerd/api/gpu/gpu-shader-module.h]: Done.

**Review by jasnell — APPROVED** [review]

**Pull request merged**